rust/sbp/src/de.rs

```diff
@@ -196,36 +189,80 @@ impl Decoder for SbpDecoder {
             }
         };
         src.advance(start);
-        match parse_frame(src) {
-            Some(Ok(frame)) => Ok(Some(Sbp::from_frame(frame)?)),
-            Some(Err(err)) => Err(err.into()),
-            None => {
-                src.reserve(MAX_FRAME_LEN);
-                Ok(None)
-            }
+        if src.len() < HEADER_LEN {
+            src.reserve(MAX_FRAME_LEN);
+            return Ok(None);
         }
+        let payload_len = src[5] as usize;
```

Comment:
maybe should extract 5 as a constant but would be kind of weird just having like `PAYLOAD_POSITION = 5` rather than being constructed via other values. HEADER_LEN - 1 ?

Reply:
Both `HEADER_LEN - 1` and a separate thing sound good to me